sentry/src/main/java/io/sentry/Span.java

```diff
@@ -193,7 +193,12 @@ public void finish(final @Nullable SpanStatus status, final @Nullable SentryDate
       @Nullable SentryDate minChildStart = null;
       @Nullable SentryDate maxChildEnd = null;
 
-      final @NotNull List<Span> children = getChildren();
+      // The root span should be trimmed based on all children, but the other spans, like the
```

Comment:
love this, explains the "why" really good and will make it easier to re-iterate over this in future.